### DISCUSSION TITLE
Module resolution

### DISCUSSION THREAD

**Discussion by grigorii-merck** [vitest-dev/vitest#2099] [Q&A]
I'm working on a project with esm enabled and for esm specifying extensions is mandatory unless using an experimental flag. However, in vitest module resolution seems different: to make things work I need to omit extensions. Can I change this behaviour somehow? Right now I'm getting an error when using a setup file. `Error: [vite-node] Failed to load ../sentry/index.js
` 
If I remove the extension, it will work in tests, but then imports break when running the server. I found a similar discussion thread but I don't use absolute paths, aliases etc.

**Comment by sheremet-va** [comment]
You don't need to omit extensions usually. Can you make a small reproduction?

**Reply by grigorii-merck** [reply]
https://github.com/grigorii-merck/vitest-module-resolution
<img width="784" alt="image" src="https://user-images.githubusercontent.com/109966561/193068116-50e2a9ed-63d9-46f8-9183-fd438705dd93.png">

**Reply by grigorii-merck** [reply]
And with omitting extension in testSetupAfterEnv.js
<img width="561" alt="image" src="https://user-images.githubusercontent.com/109966561/193069963-bc65aee3-6b7e-4167-8698-89ee8d661bef.png">

**Reply by sheremet-va** [reply]
Can you create an issue with this reproduction, please?

**Reply by grigorii-merck** [reply]
Done
